include/seqan3/range/view/translation.hpp

```diff
@@ -45,292 +45,657 @@
+ * // Second forward frame
+ * auto v5 = vec | view::translate_single(translation_frames::FWD_FRAME_1);                          // == [R,T,Y,V]
+ *
+ * // Second reverse frame
+ * auto v6 = vec | view::translate_single(translation_frames::REV_FRAME_1);                          // == [T,Y,V,R]
+ *
+ * // Third forward frame
+ * auto v7 = vec | view::translate_single(translation_frames::FWD_FRAME_2);                            // == [V,R,T]
+ *
+ * // Third reverse frame
+ * auto v8 = vec | view::translate_single(translation_frames::REV_FRAME_w);                            // == [R,T,Y]
  *
  * // function syntax
- * auto v2 = view::translate_single(vec);                      // == [T,Y,V,R]
+ * auto v9 = view::translate_single(vec, translation_frames::FWD_FRAME_0);                           // == [T,Y,V,R]
  *
  * // combinability
- * auto v3 = vec | view::complement | view::translate_single;  // == [C,M,H,A]
+ * auto v10 = vec | view::complement | view::translate_single(translation_frames::REV_FRAME_0);      // == [M,H,A,C]
+ *
+ * // combinability with default parameter
+ * auto v11 = vec | view::complement | view::translate_single;                                       // == [C,M,H,A]
+ *
+ * // combinability with default parameter
+ * auto v12 = vec | view::complement | view::translate_single();                                     // == [C,M,H,A]
+ *
  * ```
  */
-
 seqan3::detail::translate_single_fn const translate_single;
 
 } // namespace seqan3::view
 
 namespace seqan3::detail
 {
-/*!\brief The underlying type of seqan3::view::translate_frames.
+/*!\brief The return type of seqan3::view::translate.
+ * \implements seqan3::view_concept
+ * \implements seqan3::sized_range_concept
+ * \implements seqan3::random_access_range_concept
+ * \tparam urng_t The type of the range being translated.
+ * \param tf Translation frames to be used.
  * \ingroup view
  */
-struct translate_frames_fn
+template <typename urng_t>
+//!\cond
+    requires sized_range_concept<urng_t> &&
+             random_access_range_concept<urng_t> &&
+             nucleotide_concept<std::decay_t<reference_t<std::decay_t<urng_t>>>>
+//!\endcond
+class view_translate
 {
-    /*!\brief Single, reverse, forward-frame or six-frame translation from nucleotide to aminoacid.
-     * \tparam irng_t The type of the range being processed.
-     * \param irange The range being processed.
-     * \param tf Translation frames (translation_frames::SINGLE_FRAME, translation_frames::WITH_REVERSE_COMPLEMENT,
-     * translation_frames::WITH_FRAME_SHIFTS, translation_frames::SIX_FRAME).
+private:
+    //!\brief The data members of view_translate_single.
+    struct data_members_t
+    {
+        //!\brief The input range (of ranges).
+        urng_t urange;
+        //!\brief The frames that should be used for translation.
+        translation_frames const tf;
+        //!\brief The single frames corresponding to the frames required.
+        std::vector<translation_frames> single_frames{};
```

Comment:
"single_frames" may not be an ideal variable name, I first thought you actually save multiple "view::translate_single" here. Maybe "selected_frames" would be clearer?